src/QMCWaveFunctions/Fermion/DelayedUpdate.h

```diff
@@ -204,10 +206,51 @@ namespace qmcplusplus {
         else
         {
           const int lda_Binv=Binv.cols();
-          BLAS::gemm('T', 'N', delay_count, norb, norb, cone, U.data(), norb, Ainv.data(), norb, czero, tempMat.data(), lda_Binv);
-          for(int i=0; i<delay_count; i++) tempMat(delay_list[i], i) -= cone;
-          BLAS::gemm('N', 'N', norb, delay_count, delay_count, cone, V.data(), norb, Binv.data(), lda_Binv, czero, U.data(), norb);
-          BLAS::gemm('N', 'N', norb, norb, delay_count, -cone, U.data(), norb, tempMat.data(), lda_Binv, cone, Ainv.data(), norb);
+          BlasNestedThreadingService knob;
+          // always use serial when norb is small or only one second level thread
+          bool use_serial(norb<=256||knob.getNumThreads()==1);
+          if(use_serial || knob.NestedThreadingSupported())
+          {
+            // threading depends on BLAS
+            if(!use_serial) knob.presetBLASNumThreads();
```

Comment:
This looks like a good use for scoping to set/unset the BLAS threading environment.  Maybe another class (BlasThreadingEnv ?) that does RAII and takes a flag to run in serial or not?

Reply:
Good suggestion. Done.

Reply:
Thanks, it looks better now